lib/pure/times.nim

```diff
@@ -590,10 +590,28 @@ proc fromUnix*(unix: int64): Time
 
 proc toUnix*(t: Time): int64 {.benign, tags: [], raises: [], noSideEffect.} =
   ## Convert ``t`` to a unix timestamp (seconds since ``1970-01-01T00:00:00Z``).
+  ## See also `toUnix`.
   runnableExamples:
     doAssert fromUnix(0).toUnix() == 0
   t.seconds
 
+proc fromUnix*(unix: float): Time
+    {.benign, tags: [], raises: [], noSideEffect, since: (1, 1).} =
+  ## Overload working with subsecond resolution.
+  runnableExamples:
+    doAssert fromUnix(123.0) == fromUnix(123)
+  let secs = unix.floor
+  let nsecs = (unix - secs) * 1e9
+  initTime(secs.int64, nsecs.NanosecondRange)
+
+proc toUnixFloat*(t: Time): float {.benign, tags: [], raises: [].} =
```

Comment:
Needs a ``.since`` annotation. If that means code duplication for the deprecated ``toSeconds``, I don't mind. But you can easily extract the code into a helper template.

Reply:
fixed; no idea how long deprecation cycle is gonna last so code duplication is not as good